packages/headless/src/controllers/knowledge/generated-answer/headless-answerapi-generated-answer.ts

```diff
@@ -211,3 +242,128 @@ function loadAnswerApiReducers(
   engine.addReducers({[answerApi.reducerPath]: answerApi.reducer, query});
   return true;
 }
+
+const buildAdvancedSearchQueryParams = (state: StateNeededByAnswerAPI) => {
+  const advancedSearchQueryParams = selectAdvancedSearchQueries(state);
+  const mergedCq = mergeAdvancedCQParams(state);
+
+  return {
+    ...advancedSearchQueryParams,
+    ...(mergedCq && {cq: mergedCq}),
+  };
+};
+
+const mergeAdvancedCQParams = (state: StateNeededByAnswerAPI) => {
+  const activeTabExpression = selectActiveTabExpression(state.tabSet);
+  const filterExpressions = selectStaticFilterExpressions(state);
+  const {cq} = selectAdvancedSearchQueries(state);
+
+  return [activeTabExpression, ...filterExpressions, cq]
+    .filter((expression) => !!expression)
+    .join(' AND ');
+};
+
+const getNumberOfResultsWithinIndexLimit = (state: StateNeededByAnswerAPI) => {
```

Comment:
We could use the following instead: https://github.com/coveo/ui-kit/blob/4146434402013c70d353dcf59213feebf70839ba/packages/headless/src/features/search/search-request.ts#L46